### ISSUE TITLE
Connect to a device with a static pin

### ISSUE THREAD

**Issue by jhoydich** [h2zero/NimBLE-Arduino#210]
First off I wanted to thank you for the work on this library. I'm am trying to connect to a ble device that has a static 6 digit pin and I've been having some trouble connecting to it. I tried the secure client example, replacing all the necessary uuids / pincode to the appropriate values for my device. onPasskeyRequest() gets called but the connection fails just after. Any advice?

**Comment by h2zero** [comment]
Hard to say what is going on based on your description, would you be able to provide any logs?

The good news is `onPasskeyRequest()` is being called so at least we know it's getting to that point. After that, unless you are not providing the correct passkey I'm unsure of why it would fail (other than a poor connection).

**Comment by jhoydich** [comment]
Unfortunately no logs currently, would I get logs by setting the `CORE_DEBUG_LEVEL` higher?

I have quadruple checked the passcode, and I don't thing the connection is an issue, the devices are right next to each other.

**Comment by h2zero** [comment]
Yes, if you set `CORE_DEBUG_LEVEL 5` you will get the library logs. Also if you define `CONFIG_BT_NIMBLE_DEBUG` you will get additional logs from the NimBLE core.

**Comment by jhoydich** [comment]
Wonderful! Thank you! I will get some logs early next week and link here if I can't figure it out.

**Comment by jhoydich** [comment]
Turns out the specific service that I wanted was not able to be found without connecting to the device. So instead of `if (device.isAdvertisingService(serviceUuid))` I found the device based off of the device name `if (strcmp(device.getName().c_str(), "My device name") == 0)`.  Everything (almost) worked nicely then. 

The almost refers to the value returned from the characteristic. Unfortunately it returns as garbage runes instead of the expected output. Any advice? I find it odd that it prints out the device name perfectly fine but messes this up. I have attached a log showing my output and a picture of the expected output. Again, thank you for your help on this!
![IMG_4239](https://user-images.githubusercontent.com/58893325/112062843-e156f780-8b36-11eb-8451-1db4cad1220b.jpg)
[nimble_logs.txt](https://github.com/h2zero/NimBLE-Arduino/files/6185465/nimble_logs.txt)
